src/cobra/core/configuration.py

```diff
@@ -80,6 +94,12 @@ def _set_default_processes(self) -> None:
         if self.processes > 1:
             self.processes -= 1
```

Comment:
The function is only called once when the configuration is instantiated. The configuration is a singleton so it is only initialized once ever. The purpose is to set the default number of processes to `nproc - 1` so that cobrapy by default doesn't hog all CPUs. I can extend the docstring but it's not a change of this PR.